src/test/java/com/google/sps/servlets/UserDataServletTest.java

```diff
@@ -0,0 +1,256 @@
+    when(mockRequest.getParameter(USER_EMAIL_PROPERTY)).thenReturn(TEST_USER_EMAIL);
+    when(mockRequest.getParameter(USER_BIO_PROPERTY)).thenReturn(TEST_USER_BIO);
+    when(mockRequest.getParameterValues(USER_FRIENDS_LIST_PROPERTY)).thenReturn(TEST_USER_FRIENDS_LIST);
+
+    // Create a local datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // Test the doPost method by calling the wrapper method
+    servletUnderTest.doPostWrapper(datastore, mockRequest, mockResponse);
+
+    // Query the local datastore for the newly created user entity
+    PreparedQuery results = datastore.prepare(new Query(TEST_USER_ID));
+    Assert.assertEquals(1, results.countEntities());
+    Entity userEntity = results.asSingleEntity();
+
+    // Verify that the user entity has the correct properties
+    Assert.assertEquals((String) userEntity.getProperty(USER_ID_PROPERTY), TEST_USER_ID);
+    Assert.assertEquals((String) userEntity.getProperty(USER_NAME_PROPERTY), TEST_USER_NAME);
+    Assert.assertEquals((String) userEntity.getProperty(USER_EMAIL_PROPERTY), TEST_USER_EMAIL);
+    Assert.assertEquals((String) userEntity.getProperty(USER_BIO_PROPERTY), TEST_USER_BIO);
+    Assert.assertEquals(((ArrayList<String>) userEntity.getProperty(USER_FRIENDS_LIST_PROPERTY)).toArray(),
+        TEST_USER_FRIENDS_LIST);
+  }
+
+  /**
+   * Tests the doPost method, making sure that a user entity is updated in Datastore.
+   * Expected response: Updates the user entity already in the local datastore, and
+   * changes the user's name from "Tim" to "John".
+   */
+  @Test
+  public void testPostUpdateUserInfo() throws Exception {
+    // Mock the getParameter and getParameterValues calls
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+    when(mockRequest.getParameter(USER_NAME_PROPERTY)).thenReturn(ALTERNATE_TEST_USER_NAME);
+    when(mockRequest.getParameter(USER_EMAIL_PROPERTY)).thenReturn(TEST_USER_EMAIL);
+    when(mockRequest.getParameter(USER_BIO_PROPERTY)).thenReturn(TEST_USER_BIO);
+    when(mockRequest.getParameterValues(USER_FRIENDS_LIST_PROPERTY)).thenReturn(TEST_USER_FRIENDS_LIST);
+
+    // Create a local datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    addTestUserEntityToDatastore(datastore);
+    // Test the doPost method by calling the wrapper method
+    servletUnderTest.doPostWrapper(datastore, mockRequest, mockResponse);
+
+    // Query the local datastore for the newly created user entity
+    PreparedQuery results = datastore.prepare(new Query(TEST_USER_ID));
+    Assert.assertEquals(1, results.countEntities());
+    Entity userEntity = results.asSingleEntity();
+
+    // Verify that the user Entity name property changed
+    Assert.assertEquals((String) userEntity.getProperty(USER_ID_PROPERTY), TEST_USER_ID);
+    Assert.assertEquals((String) userEntity.getProperty(USER_NAME_PROPERTY), ALTERNATE_TEST_USER_NAME);
```

Comment:
This is really the only line you need here. You tested all the other lines in the other test case. You can leave the ID check if you want (actually I'd argue that's a good idea).

Reply:
I think I'll leave that ID check there, just to test one of the properties that are not changing. Also because entities `kind`s are their user ids.